### ISSUE TITLE
4.10.7.2.12 The placeholder attribute

### ISSUE THREAD

**Issue by devarshipant** [w3c/html#1014]
4.10.7.2.12 The placeholder attribute:

http://www.w3.org/TR/2011/WD-html5-author-20110809/spec.html#attr-input-placeholder

The following two comments should be revisited in light of #D (accessible name calculation):

> For a longer hint or other advisory text, the title attribute is more appropriate.

> The placeholder attribute should not be used as an alternative to a label.

Accessible Name calculation (http://www.w3.org/TR/accname-aam-1.1/#mapping_additional_nd_te)

Refer to #D where is specifically states "Otherwise, if the current node's native markup provides an attribute or element that defines a text alternative, return that alternative as a flat string, unless the element is marked as presentational (role="presentation" or role="none").

**Comment by LJWatson** [comment]
Thanks @devarshipant.

You've referenced a version of HTML5 from 2011. The ]HTML5.2 `@placeholder` definition](http://w3c.github.io/html/sec-forms.html#the-placeholder-attribute) no longer includes the suggestion to use the `@title` attribute.

The advice not to use `@placeholder` as a substitute for the ```<label>``` element still seems relevant though, for the reasons described in HTML5.2.

**Comment by devarshipant** [comment]
Thanks for the feedback @LJWatson -- So what happens when the text alternative computation (item #D)calculates the accessible name using the placeholder attribute when it is the only attribute (in the absence of valid label etc.)? Should there be guidance in HTML 5.2 that recommends not to rely on the accessible name when just the placeholder is in use?

**Comment by LJWatson** [comment]
>" -- So what happens when the text alternative computation (item #D)calculates the accessible name using the placeholder attribute when it is the only attribute
(in the absence of valid label etc.)? Should there be guidance in HTML 5.2 that recommends not to rely on the accessible name when just the placeholder
is in use?">

That sounds like something that would need to be handled in the accessible name computation, not the HTML spec. The HTML spec advises against a situation where placeholder would constitute the accessible name for a field, I'm not sure it can advise against the placeholder being the accessible name in the absence of any alternative though.

**Comment by devarshipant** [comment]
Thanks @LJWatson -- I would assume folks who manage accessible name computation algorithm should insert an advisory in API mappings to capture this.

**Comment by LJWatson** [comment]
@devarshipant
 yes, I think so. You might want to file an issue against the [ARIA repo](https://github.com/w3c/aria/) where the Acc Name Computation spec is produced.
